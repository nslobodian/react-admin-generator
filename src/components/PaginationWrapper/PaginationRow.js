import React from 'react'
import CountFilter from './CountFilter'
import Pagination from 'react-js-pagination'
import s from './PaginationWrapper.scss'
import Dropdown from 'react-select'
import pure from 'recompose/pure'

export const DropdownWrapper = pure(({ options, toggleChangeHandle, value, dropdownConfig = {} }) =>
  <Dropdown options={options} value={value}
    clearable={false} onChange={toggleChangeHandle}
    labelKey={dropdownConfig.labelKey || 'label'}
    valueKey={dropdownConfig.valueKey || 'value'}
/>)

export const PaginationRow = ({ perPage, pages, page, handleChange, totalCount, label, options, toggleChangeHandle, perUser, short, dropdownConfig, dropdownValue }) =>
  <div className='row'>
    {!!options &&
    <div className='col-sm-3 col-xs-6'>
      <div className={s.additionalComponentWrapper}>
        <DropdownWrapper options={options} toggleChangeHandle={toggleChangeHandle} value={dropdownConfig ? dropdownValue : options[perUser ? 0 : 1]} dropdownConfig={dropdownConfig} />
      </div>
    </div>
    }
    <div className={`col-lg-6 col-sm-${short ? '6' : '4'} col-xs-6`}>
      <CountFilter perPage={perPage} total={totalCount} onChangeFilter={handleChange} label={label} short={short} />
    </div>
    <div className={`col-lg-${options ? '3' : '6'} col-sm-${options ? '5' : short ? '6' : '8'} col-xs-12`}>
      <div className={s.Pagination}>
        <Pagination
          activePage={page}
          itemsCountPerPage={perPage}
          totalItemsCount={totalCount || 0}
          pageRangeDisplayed={short ? 2 : 5}
          linkClass={short ? 'smallPagination' : ''}
          onChange={val => handleChange({ page: val })}
        />
      </div>
    </div>
  </div>

export default PaginationRow
