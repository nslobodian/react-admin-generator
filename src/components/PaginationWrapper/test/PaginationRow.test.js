import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { DropdownWrapper, PaginationRow } from '../PaginationRow'

describe('(Components) PaginationRow', () => {
  it('Should render DropdownWrapper properly without dropdownConfig', () => {
    const options = [{
      filter: jest.fn(),
    }];
    const toggleChangeHandle = jest.fn();
    const value = 'value';
    const wrapper = shallow(
      <DropdownWrapper
        value={value}
        toggleChangeHandle={toggleChangeHandle}
        options={options}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render DropdownWrapper properly with dropdownConfig', () => {
    const options = [{
      labelKey: 'labelKey',
      valueKey: 'valueKey',
    }];
    const dropdownConfig = [{
      filter: jest.fn(),
    }];
    const toggleChangeHandle = jest.fn();
    const value = 'value';
    const wrapper = shallow(
      <DropdownWrapper
        value={value}
        toggleChangeHandle={toggleChangeHandle}
        options={options}
        dropdownConfig={dropdownConfig}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render PaginationRow properly', () => {
    const perPage = 10;
    const pages = 1;
    const page = 1;
    const handleChange = jest.fn();
    const totalCount = 1;
    const label = 'label';
    const toggleChangeHandle = jest.fn();
    const perUser = 1;
    const short = false;
    const dropdownValue = true;
    const options = [{
      labelKey: 'labelKey',
      valueKey: 'valueKey',
    }];
    const dropdownConfig = [{
      filter: jest.fn(),
    }];
    const wrapper = shallow(
      <PaginationRow
        perPage={perPage}
        pages={pages}
        page={page}
        handleChange={handleChange}
        totalCount={totalCount}
        label={label}
        options={options}
        toggleChangeHandle={toggleChangeHandle}
        perUser={perUser}
        short={short}
        dropdownValue={dropdownValue}
        dropdownConfig={dropdownConfig}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render PaginationRow properly with no options and short === false', () => {
    const perPage = 10;
    const pages = 1;
    const page = 1;
    const handleChange = jest.fn();
    const totalCount = 1;
    const label = 'label';
    const toggleChangeHandle = jest.fn();
    const perUser = true;
    const short = false;
    const dropdownValue = true;
    const options = false;
    const dropdownConfig = [{
      filter: jest.fn(),
    }];
    const wrapper = shallow(
      <PaginationRow
        perPage={perPage}
        pages={pages}
        page={page}
        handleChange={handleChange}
        totalCount={totalCount}
        label={label}
        options={options}
        toggleChangeHandle={toggleChangeHandle}
        perUser={perUser}
        short={short}
        dropdownValue={dropdownValue}
        dropdownConfig={dropdownConfig}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render PaginationRow properly with opposite values', () => {
    const perPage = 10;
    const pages = 1;
    const page = 1;
    const handleChange = jest.fn();
    const totalCount = 1;
    const label = 'label';
    const toggleChangeHandle = jest.fn();
    const perUser = 1;
    const short = true;
    const dropdownValue = true;
    const options = false;
    const dropdownConfig = [{
      filter: jest.fn(),
    }];
    const wrapper = shallow(
      <PaginationRow
        perPage={perPage}
        pages={pages}
        page={page}
        handleChange={handleChange}
        totalCount={totalCount}
        label={label}
        options={options}
        toggleChangeHandle={toggleChangeHandle}
        perUser={perUser}
        short={short}
        dropdownValue={dropdownValue}
        dropdownConfig={dropdownConfig}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
