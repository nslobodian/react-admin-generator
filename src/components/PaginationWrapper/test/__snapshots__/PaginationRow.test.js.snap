// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) PaginationRow Should render DropdownWrapper properly with dropdownConfig 1`] = `
<Select
  addLabelText="Add \\"{label}\\"?"
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={false}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "labelKey": "labelKey",
        "valueKey": "valueKey",
      },
    ]
  }
  pageSize={5}
  placeholder="Select..."
  required={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  value="value"
  valueComponent={[Function]}
  valueKey="value"
/>
`;

exports[`(Components) PaginationRow Should render DropdownWrapper properly without dropdownConfig 1`] = `
<Select
  addLabelText="Add \\"{label}\\"?"
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={false}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "filter": [Function],
      },
    ]
  }
  pageSize={5}
  placeholder="Select..."
  required={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  value="value"
  valueComponent={[Function]}
  valueKey="value"
/>
`;

exports[`(Components) PaginationRow Should render PaginationRow properly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-3 col-xs-6"
  >
    <div
      className="additionalComponentWrapper"
    >
      <_class
        dropdownConfig={
          Array [
            Object {
              "filter": [Function],
            },
          ]
        }
        options={
          Array [
            Object {
              "labelKey": "labelKey",
              "valueKey": "valueKey",
            },
          ]
        }
        toggleChangeHandle={[Function]}
        value={true}
      />
    </div>
  </div>
  <div
    className="col-lg-6 col-sm-4 col-xs-6"
  >
    <_class
      label="label"
      onChangeFilter={[Function]}
      perPage={10}
      short={false}
      total={1}
    />
  </div>
  <div
    className="col-lg-3 col-sm-5 col-xs-12"
  >
    <div
      className="Pagination"
    >
      <Pagination
        activePage={1}
        firstPageText="«"
        innerClass="pagination"
        itemsCountPerPage={10}
        lastPageText="»"
        linkClass=""
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={5}
        prevPageText="⟨"
        totalItemsCount={1}
      />
    </div>
  </div>
</div>
`;

exports[`(Components) PaginationRow Should render PaginationRow properly with no options and short === false 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6 col-sm-4 col-xs-6"
  >
    <_class
      label="label"
      onChangeFilter={[Function]}
      perPage={10}
      short={false}
      total={1}
    />
  </div>
  <div
    className="col-lg-6 col-sm-8 col-xs-12"
  >
    <div
      className="Pagination"
    >
      <Pagination
        activePage={1}
        firstPageText="«"
        innerClass="pagination"
        itemsCountPerPage={10}
        lastPageText="»"
        linkClass=""
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={5}
        prevPageText="⟨"
        totalItemsCount={1}
      />
    </div>
  </div>
</div>
`;

exports[`(Components) PaginationRow Should render PaginationRow properly with opposite values 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6 col-sm-6 col-xs-6"
  >
    <_class
      label="label"
      onChangeFilter={[Function]}
      perPage={10}
      short={true}
      total={1}
    />
  </div>
  <div
    className="col-lg-6 col-sm-6 col-xs-12"
  >
    <div
      className="Pagination"
    >
      <Pagination
        activePage={1}
        firstPageText="«"
        innerClass="pagination"
        itemsCountPerPage={10}
        lastPageText="»"
        linkClass="smallPagination"
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={2}
        prevPageText="⟨"
        totalItemsCount={1}
      />
    </div>
  </div>
</div>
`;
